src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration/ConfigurationBinderBenchmarks.cs

```diff
@@ -0,0 +1,91 @@
+﻿using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.Extensions.Configuration
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class ConfigurationBinderBenchmarks
+    {
+        public static IEnumerable<int> GetDuplicateCount()
+            => Enumerable.Range(1, 3).Select(i => 2 * i);
+
+        public static IEnumerable<int> GetMyObjectCount()
+            => Enumerable.Range(1, 3).Select(i => 3 * i);
+
+        [ParamsSource(nameof(GetMyObjectCount))]
+        public int MyObjectCount { get; set; }
+
+        [ParamsSource(nameof(GetDuplicateCount))]
+        public int DuplicateCount { get; set; }
+
+        private IConfiguration _configuration;
+
+        private readonly List<MemoryStream> _streams = new List<MemoryStream>();
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            var builder = new ConfigurationBuilder();
+            for (int i = 0; i < this.MyObjectCount; i++)
+            {
+                var s = new MySettings
+                {
+                    Foo = "root",
+                    Bar = int.MaxValue,
+                    SubSettings = new Dictionary<string, MySettings>
+                    {
+                        ["level 1 key " + i] = new MySettings
+                        {
+                            Foo = "foo" + i,
+                            Bar = i,
+                            SubSettings = new Dictionary<string, MySettings>
+                            {
+                                ["level 2 key " + i] = new MySettings
+                                {
+                                    Foo = "just a simple 2 level tree settings " + i,
+                                    Bar = 2 * i,
+                                }
+                            }
+                        }
+                    }
+                };
+                var jsonString = JsonConvert.SerializeObject(s);
+                for (var j = 0; j < DuplicateCount; j++)
+                {
+                    var stream = new MemoryStream(Encoding.UTF8.GetBytes(jsonString));
+                    this._streams.Add(stream);
+                    builder.AddJsonStream(stream);
+                }
+            }
+
+            _configuration = builder.Build();            
+        }
+
+        [GlobalCleanup]
+        public void Cleanup()
+        {
+            foreach (var s in this._streams)
+            {
+                s.Dispose();
```

Comment:
Technically, MemoryStreams don't need to be disposed, since they don't wrap resources that need to be cleaned up - only managed memory which will be GC'd. So you can remove this, and the `List<MemoryStream> _streams` field.

See the "Note" in the Remarks section https://docs.microsoft.com/en-us/dotnet/api/system.io.memorystream?view=net-6.0#remarks

Reply:
Thanks for your suggestion. I have removed clean up method and _streams field.